step2/hackatom-stash-controller.sh

```diff
@@ -1,79 +1,106 @@
 #!/usr/bin/env bash
 set -e
 
-# This script is for setting up the stash and controller accounts.
+# This script is for setting up the stash and controller accounts, and sending
+# some txs from controller on behalf of stash. All commands are commented.
 
 # Set variables
 CFG_DIR=~/.simapp
-BUILD_CMD=./cosmos-sdk/build/simd
+BUILD_CMD=../step1/cosmos-sdk/build/simd
 VALIDATOR=alice
 CHAIN_ID=hackatom6-chain
+MIN_FEES=20stake
 
 # Key names in keyring
 STASH=stash
 CONTROLLER=controller
 
 # Get validator address
-validator=$("${BUILD_CMD}" keys show $VALIDATOR --bech val --keyring-backend test --output json)
-VALIDATOR_ADDRESS=$(echo "${validator}" | jq -r '.address')
+VALIDATOR_ADDRESS=$($BUILD_CMD keys show $VALIDATOR --bech val -a --keyring-backend test)
 
-# Create stash account in keyring.
+echo
+echo "Create stash account in keyring."
 $BUILD_CMD keys add $STASH --keyring-backend test 
 STASH_ADDRESS=$($BUILD_CMD keys show $STASH -a --keyring-backend test)
 
-# Create controller account in keyring.
+echo
+echo "Create controller account in keyring."
 $BUILD_CMD keys add $CONTROLLER --keyring-backend test
 CONTROLLER_ADDRESS=$($BUILD_CMD keys show $CONTROLLER -a --keyring-backend test)
 
-# Send a lot of funds to STASH.
-$BUILD_CMD tx bank send $VALIDATOR $STASH_ADDRESS 10000000stake --keyring-backend test --chain-id $CHAIN_ID --broadcast-mode block -y
+echo
+echo "Send a lot of funds to STASH."
+$BUILD_CMD tx bank send $VALIDATOR $STASH_ADDRESS 10000000stake --keyring-backend test --chain-id $CHAIN_ID --broadcast-mode block --fees $MIN_FEES -y
 
-# Send funds to CONTROLLER from $STASH for fee purposes.
-$BUILD_CMD tx bank send $STASH $CONTROLLER_ADDRESS 100000stake --keyring-backend test --chain-id $CHAIN_ID --broadcast-mode block -y
```

Comment:
I removed this. The whole point of feegrant is that CONTROLLER doesn't need any funds.

See the usage of `--fee-account` below